src/ristretto/pedersen/extended_commitment_factory.rs

```diff
@@ -0,0 +1,415 @@
+
+impl ExtendedPedersenCommitmentFactory {
+    /// Create a new Extended Pedersen Ristretto Commitment factory for the required extension degree using
+    /// pre-calculated compressed constants - we only hold references to the static generator points.
+    pub fn new_with_extension_degree(extension_degree: ExtensionDegree) -> Result<Self, RangeProofError> {
+        if extension_degree as usize > RISTRETTO_NUMS_POINTS.len() ||
+            extension_degree as usize > RISTRETTO_NUMS_POINTS_COMPRESSED.len()
+        {
+            return Err(RangeProofError::ExtensionDegree(
+                "Not enough Ristretto NUMS points to construct the extended commitment factory".to_string(),
+            ));
+        }
+        let mut g_base_vec = Vec::with_capacity(extension_degree as usize);
+        g_base_vec.push(RISTRETTO_PEDERSEN_G);
+        let mut g_base_compressed_vec = Vec::with_capacity(extension_degree as usize);
+        g_base_compressed_vec.push(RISTRETTO_PEDERSEN_G_COMPRESSED);
+        for i in 1..extension_degree as usize {
+            g_base_vec.push(RISTRETTO_NUMS_POINTS[i]);
+            g_base_compressed_vec.push(RISTRETTO_NUMS_POINTS_COMPRESSED[i]);
+        }
+        Ok(Self(PedersenGens {
+            h_base: *RISTRETTO_PEDERSEN_H,
+            h_base_compressed: *RISTRETTO_PEDERSEN_H_COMPRESSED,
+            g_base_vec,
+            g_base_compressed_vec,
+            extension_degree,
+        }))
+    }
+}
+
+impl Default for ExtendedPedersenCommitmentFactory {
+    /// The default Extended Pedersen Ristretto Commitment factory is of extension degree Zero; this corresponds to
+    /// the default non extended Pedersen Ristretto Commitment factory.
+    fn default() -> Self {
+        Self::new_with_extension_degree(ExtensionDegree::Zero).expect("Ristretto default base points not defined!")
+    }
+}
+
+impl ExtendedHomomorphicCommitmentFactory for ExtendedPedersenCommitmentFactory {
+    type P = RistrettoPublicKey;
+
+    fn commit(
+        &self,
+        k_i: &[RistrettoSecretKey],
+        v: &RistrettoSecretKey,
+    ) -> Result<PedersenCommitment, RangeProofError> {
+        let k_i: Vec<Scalar> = k_i.to_vec().iter().map(|k| k.0).collect();
+        let c = self
+            .0
+            .commit(v.0, &k_i)
+            .map_err(|e| RangeProofError::ExtensionDegree(e.to_string()))?;
+        Ok(HomomorphicCommitment(RistrettoPublicKey::new_from_pk(c)))
+    }
+
+    fn zero(&self) -> PedersenCommitment {
+        let extension_degree = self.0.extension_degree as usize;
+        let zero = vec![Scalar::zero(); extension_degree + 1];
+        let mut points = Vec::with_capacity(extension_degree + 1);
+        points.push(self.0.h_base);
+        points.append(&mut self.0.g_base_vec.clone());
+        let c = RistrettoPoint::multiscalar_mul(&zero, &self.0.g_base_vec);
+        HomomorphicCommitment(RistrettoPublicKey::new_from_pk(c))
+    }
```

Comment:
This must always return the group identity, so there's no need to compute this multiscalar multiplication.

Reply:
Updated and added a specific test case for it